src/interface.jl

```diff
@@ -0,0 +1,252 @@
+    error("`logabsdetjac(b::$T1, y::$T2)` is not implemented.")
+
+"Transforms the input using the bijector."
+transform(b::T1, y::T2) where {T<:Bijector,T1<:Inversed{T},T2} =
+    error("`transform(b::$T1, y::$T2)` is not implemented.")
+
+"Computes both `transform` and `logabsdetjac` in one forward pass."
+forward(b::T1, y::T2) where {T<:Bijector,T1<:Inversed{T},T2} = 
+    error("`forward(b::$T1, y::$T2)` is not implemented.")
+
+
+transform(b::Bijector) = x -> transform(b, x)
+(ib::Inversed{<: Bijector})(y) = transform(ib, y)
+
+# default `forward` implementations; should in general implement efficient way
+# of computing both `transform` and `logabsdetjac` together.
+forward(b::Bijector, x) = (rv=transform(b, x), logabsdetjac=logabsdetjac(b, x))
+forward(ib::Inversed{<: Bijector}, y) = (rv=transform(ib, y), logabsdetjac=logabsdetjac(ib, y))
+
+# defaults implementation for inverses
+logabsdetjac(ib::Inversed{<: Bijector}, y) = - logabsdetjac(ib.orig, transform(ib, y))
+
+inv(b::Bijector) = Inversed(b)
+inv(ib::Inversed{<:Bijector}) = ib.orig
+
+# AD implementations
+function jacobian(b::ADBijector{<: ForwardDiffAD}, y::Real)
+    ForwardDiff.derivative(z -> transform(b, z), y)
+end
+function jacobian(b::Inversed{<: ADBijector{<: ForwardDiffAD}}, y::Real)
+    ForwardDiff.derivative(z -> transform(b, z), y)
+end
+function jacobian(b::Inversed{<: ADBijector{<: ForwardDiffAD}}, y::AbstractVector{<: Real})
+    ForwardDiff.jacobian(z -> transform(b, z), y)
+end
+
+function jacobian(b::ADBijector{<: TrackerAD}, y::Real)
+    Tracker.gradient(z -> transform(b, z), y)[1]
+end
+function jacobian(b::Inversed{<: ADBijector{<: TrackerAD}}, y::Real)
+    Tracker.gradient(z -> transform(b, z), y)[1]
+end
+function jacobian(b::Inversed{<: ADBijector{<: TrackerAD}}, y::AbstractVector{<: Real})
+    Tracker.jacobian(z -> transform(b, z), y)
+end
+
+# TODO: allow batch-computation, especially for univariate case?
+"Computes the absolute determinant of the Jacobian of the inverse-transformation."
+function logabsdetjac(b::ADBijector, x::Real)
+    log(abs(jacobian(b, x)))
+end
+function logabsdetjac(b::ADBijector, x::AbstractVector{<:Real})
+    fact = lu(jacobian(b, x), check=false)
+    issuccess(fact) ? log(abs(det(fact))) : -Inf # TODO: or smallest possible float?
+end
+
+###############
+# Composition #
+###############
+
+struct Composed{A} <: Bijector
+    ts::A
+end
+
+compose(ts...) = Composed(ts)
+
+# The transformation of `Composed` applies functions left-to-right
+# but in mathematics we usually go from right-to-left; this reversal ensures that
+# when we use the mathematical composition ∘ we get the expected behavior.
+# TODO: change behavior of `transform` of `Composed`?
+∘(b1::B1, b2::B2) where {B1 <: Bijector, B2 <: Bijector} = compose(b2, b1)
+
+inv(ct::Composed) = Composed(map(inv, reverse(ct.ts)))
+
+# TODO: can we implement this recursively, and with aggressive inlining, make this type-stable?
```

Comment:
Probably want to dispatch for this. If the container is a tuple, then recursion + inlining is probably a good move. If it's a vector, then probably not.

Reply:
I just tried the following:
```julia
_transform(x, b1::Bijector, b2::Bijector) = transform(b2, transform(b1, x))
_transform(x, b::Bijector, bs::Bijector...) = _transform(transform(b, x), bs)  # note ordered differently from above
transform(cb::Composed{<: Tuple}, x) = _transform(x, cb.ts...)
```
And this seems to do the trick!